geode-core/src/test/java/org/apache/geode/management/internal/cli/commands/DeployCommandTest.java

```diff
@@ -57,4 +57,10 @@ public void bothDirAndJar() {
   public void missingDirOrJar() {
     gfsh.executeAndAssertThat(command, "deploy").statusIsError().containsOutput("is required");
   }
+
+  @Test
+  public void convertShorthandToPathJar() {
+    gfsh.executeAndAssertThat(command, "deploy --jar=~/abc.jar").statusIsError()
```

Comment:
Add a test that `--jar=~foo/bar.jar` and `--jar=foo~/bar.jar` are not modified.

Reply:
Technically (at least considering shell semantics) `~foo` is legitimate (it expands to the home directory of the `foo` user if that user exists) but I don't think we should attempt to expand that.